File: src/main/java/com/snowofsunflower/android/ui/list/ItemViewHolder.java
package com.snowofsunflower.android.ui.list;

import android.support.annotation.IdRes;
import android.view.View;

/**
 * Created by zhouztashin on 2018/10/31.
 */

public interface ItemViewHolder {
    <T extends View> T findView(@IdRes int id);

    int getPosition();

    ItemViewHolder setText(@IdRes int viewId, CharSequence value);
    ItemViewHolder setGone(@IdRes int viewId);
    ItemViewHolder setVisible(@IdRes int viewId);
    ItemViewHolder setInvisible(@IdRes int viewId);
}

File: src/main/java/com/snowofsunflower/android/ui/list/ItemViewHolder.kt
package com.snowofsunflower.android.ui.list

import android.view.View
import androidx.annotation.IdRes

/**
 * Created by zhouztashin on 2018/10/31.
 * 列表项目处理
 */

interface ItemViewHolder {

    val position: Int
    fun <T : View> findView(@IdRes id: Int): T
    fun setText(@IdRes viewId: Int, value: CharSequence): ItemViewHolder
    fun setGone(@IdRes viewId: Int): ItemViewHolder
    fun setVisible(@IdRes viewId: Int): ItemViewHolder
    fun setInvisible(@IdRes viewId: Int): ItemViewHolder
}
